encoding/iid/VertexIID.java

```diff
@@ -233,44 +231,30 @@ private Attribute(ByteArray bytes) {
         }
 
         public static VertexIID.Attribute<?> of(ByteArray bytes) {
-            switch (Encoding.ValueType.of(bytes.get(PREFIX_W_TYPE_LENGTH))) {
-                case BOOLEAN:
-                    return new Attribute.Boolean(bytes);
-                case LONG:
-                    return new Attribute.Long(bytes);
-                case DOUBLE:
-                    return new Attribute.Double(bytes);
-                case STRING:
-                    return new Attribute.String(bytes);
-                case DATETIME:
-                    return new Attribute.DateTime(bytes);
-                default:
-                    assert false;
-                    throw TypeDBException.of(UNRECOGNISED_VALUE);
-            }
+            Encoding.ValueType<?> valueType = Encoding.ValueType.of(bytes.get(PREFIX_W_TYPE_LENGTH));
+            if (valueType == BOOLEAN) return new Boolean(bytes);
+            else if (valueType == LONG) return new Long(bytes);
+            else if (valueType == DOUBLE) return new Double(bytes);
+            else if (valueType == STRING) return new String(bytes);
+            else if (valueType == DATETIME) return new DateTime(bytes);
+            assert false;
+            throw TypeDBException.of(UNRECOGNISED_VALUE);
         }
 
         public static VertexIID.Attribute<?> extract(ByteArray bytes, int from) {
-            switch (Encoding.ValueType.of(bytes.get(from + VALUE_TYPE_INDEX))) {
-                case BOOLEAN:
-                    return VertexIID.Attribute.Boolean.extract(bytes, from);
-                case LONG:
-                    return VertexIID.Attribute.Long.extract(bytes, from);
-                case DOUBLE:
-                    return VertexIID.Attribute.Double.extract(bytes, from);
-                case STRING:
-                    return VertexIID.Attribute.String.extract(bytes, from);
-                case DATETIME:
-                    return VertexIID.Attribute.DateTime.extract(bytes, from);
-                default:
-                    assert false;
-                    throw TypeDBException.of(UNRECOGNISED_VALUE);
-            }
+            Encoding.ValueType<?> valueType = Encoding.ValueType.of(bytes.get(from + VALUE_TYPE_INDEX));
+            if (valueType == BOOLEAN) return Boolean.extract(bytes, from);
+            else if (valueType == LONG) return Long.extract(bytes, from);
+            else if (valueType == DOUBLE) return Double.extract(bytes, from);
+            else if (valueType == STRING) return String.extract(bytes, from);
+            else if (valueType == DATETIME) return DateTime.extract(bytes, from);
+            assert false;
+            throw TypeDBException.of(UNRECOGNISED_VALUE);
         }
 
         public abstract VALUE value();
 
-        public Encoding.ValueType valueType() {
+        public Encoding.ValueType<VALUE> valueType() {
```

Comment:
the parameter here being a not `?` type allows us to use it for operations elsewhere